### PR TITLE
feat(autoware_auto_perception_rviz_plugin): improve rviz performance

### PR DESCRIPTION
**Pull request by lexavtanke**
## Description

This PR add ability to choose between different types of DetectedObjects visualization: 3d polygon, 2d polygon, no polygon.
Here is [video](https://youtu.be/bLkYjFaqPzs) showing this. 

There is also small performance improvement with big amount of objects (1-2 fps). 
Here are [250 detected objects](https://youtu.be/8i9m_NrQFwU).
Here are [500](https://youtu.be/gYQVqmL4Eqk).

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by taikitanaka3** [comment]
@yukkysaito 
Can you review this PR?

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp [hunk 1]
```diff
@@ -435,6 +516,37 @@ void calc_cylinder_line_list(
...
+                  M_PI / static_cast<double>(n)) *
+                radius;
+      point.y = std::sin(
+                  (static_cast<double>(i) / static_cast<double>(n)) * 2.0 * M_PI +
+                  M_PI / static_cast<double>(n)) *
+                radius;
+      point.z = -shape.dimensions.z * 0.5;
+      points.push_back(point);
+    }
+  }
```
Comment (yukkysaito) [inline common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp:547]: I think this is unnecessary code.

**Comment by yukkysaito** [comment]
@lexavtanke Thank you for PR :+1: I reviewed the code

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp [hunk 2]
```diff
@@ -392,6 +430,49 @@ void calc_bounding_box_line_list(
   points.push_back(point);
 }
 
+void calc_2d_bounding_box_line_list(
```
Comment (yukkysaito) [inline common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp:433]:
```suggestion
void calc_2d_bounding_box_bottom_line_list(
```
maybe better to use the word (`bottom`)

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp [hunk 3]
```diff
@@ -435,6 +516,37 @@ void calc_cylinder_line_list(
   }
 }
 
+void calc_cylinder_bottom_line_list(
```
Comment (yukkysaito) [inline common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp:519]:
```suggestion
void calc_2d_cylinder_bottom_line_list(
```

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp [hunk 4]
```diff
@@ -521,6 +633,33 @@ void calc_polygon_line_list(
   }
 }
 
+void calc_2d_polygon_line_list(
```
Comment (yukkysaito) [inline common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp:636]:
```suggestion
void calc_2d_polygon_bottom_line_list(
```

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp (hunk 1 above)
Comment (lexavtanke) [reply common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp]: deleted

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp (hunk 2 above)
Comment (lexavtanke) [reply common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp]: changed

File: common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp (hunk 4 above)
Comment (lexavtanke) [reply common/autoware_auto_perception_rviz_plugin/src/object_detection/object_polygon_detail.cpp]: changed

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**